source/adapters/level_zero/context.hpp

```diff
@@ -225,17 +245,29 @@ struct ur_context_handle_t_ : _ur_object {
     HostVisibleCounterBasedRegularCacheType,
     HostInvisibleCounterBasedRegularCacheType,
     HostVisibleCounterBasedImmediateCacheType,
-    HostInvisibleCounterBasedImmediateCacheType
+    HostInvisibleCounterBasedImmediateCacheType,
+
+    HostVisibleInterruptBasedRegularCacheType,
+    HostInvisibleInterruptBasedRegularCacheType,
+    HostVisibleInterruptBasedImmediateCacheType,
+    HostInvisibleInterruptBasedImmediateCacheType,
+
+    HostVisibleInterruptAndCounterBasedRegularCacheType,
+    HostInvisibleInterruptAndCounterBasedRegularCacheType,
+    HostVisibleInterruptAndCounterBasedImmediateCacheType,
+    HostInvisibleInterruptAndCounterBasedImmediateCacheType
```

Comment:
This is exactly why I've asked you to do the refactor. This will grow exponentially with new event parameters.

Reply:
I agree, @winstonzhang-intel has https://github.com/oneapi-src/unified-runtime/pull/1600 for the refactor that has been moved back due to other issues, but I have asked him to finish that this week with your recent changes to the cache.

I think the refactor though should be in that refactor PR after this change to avoid over complicating this feature review.

Reply:
Since there will be some refactoring done with this PR, I will work on the refactor patch after this is in. https://github.com/oneapi-src/unified-runtime/pull/1600